### PR TITLE
Add Show Tips toggle

### LINKED ISSUE #7039: NUX: Allow user to re-enable tips
From https://github.com/WordPress/gutenberg/pull/6631#issuecomment-393069320:

> I like tips. In video games tips are usually togglable in an Options menu. Should we make an option in the More menu that is enabled by default for new users, and is disabled when you're done or dismiss it?

Let's definitely do this. Quick mockup:

screen shot 2018-05-31 at 16 25 15

cc. @karmatosed @jasmussen

### PR DESCRIPTION
**Pull request by noisysocks**
Closes #7039. 

tips-toggle

Lets the user re-enable tips via a Show Tips toggle in the More menu. Re-enabling tips will also make previously dismissed tips appear again.

Since there's way to get tips back, I've made the X button in the tip popup disable all tips instead of dismissing the current tip.

## How has this been tested?

1. Create a new post
2. Step through or dismiss the guide if it appears
3. Re-enable tips via the More menu
4. Disable tips via the More menu

### CODE REVIEW

**Comment by noisysocks** [comment]
Whaddya reckon, @karmatosed?

**Review by karmatosed — APPROVED** [review]
From a design perspective approving. I think this is a great improvement to Tips.

**Comment by noisysocks** [comment]
Thanks Tammie! Tagged some suggested developers for a code review 🙂

**Review by aduth — COMMENTED** [review]
Observation: When I first loaded up the branch, the menu item showed as checked, yet I did not see any tip prompts. I had to disable the tips and then reenable them once more for them to be displayed. Is this more a consequence of some lingering localStorage, or is it possible a user would find themselves in this situation (not the most obvious to work their way out of).

File: edit-post/components/header/fixed-toolbar-toggle/index.js [hunk 1]
```diff
@@ -8,23 +8,18 @@ import { withSelect, withDispatch } from '@wordpress/data';
  */
 import { __ } from '@wordpress/i18n';
 import { compose } from '@wordpress/element';
-import { MenuGroup, MenuItem, withInstanceId } from '@wordpress/components';
+import { MenuItem } from '@wordpress/components';
 import { ifViewportMatches } from '@wordpress/viewport';
 
-function FeatureToggle( { onToggle, isActive } ) {
+function FixedToolbarToggle( { onToggle, isActive } ) {
```
Comment (aduth) [inline edit-post/components/header/fixed-toolbar-toggle/index.js]: While an opposing argument may be made that the unrelated changes are a distraction to the review, I appreciate the small refactorings made in passing here between the component name correction and dropping the unused `withInstanceId`. ❤️

File: edit-post/components/header/more-menu/index.js [hunk 2]
```diff
@@ -27,7 +28,13 @@ const MoreMenu = () => (
...
 			<div className="edit-post-more-menu__content">
 				<ModeSwitcher onSelect={ onClose } />
-				<FixedToolbarToggle onToggle={ onClose } />
+				<MenuGroup
+					label={ __( 'Settings' ) }
+					filterName="editPost.MoreMenu.settings"
+				>
+					<FixedToolbarToggle onToggle={ onClose } />
+					<TipsToggle onToggle={ onClose } />
+				</MenuGroup>
```
Comment (mcsf) [inline edit-post/components/header/more-menu/index.js]: This highlights that we have diverging interaction patterns within MoreMenu:
- Fix Toolbar to Top: click triggers action and closes menu
- Show Tips: _idem_
- Copy All Content: click triggers action, success is shown in item label, menu remains open

That said, for the purpose of this PR, we can move on. Just something to be aware of.

**Review by mcsf — APPROVED** [review]

**Review by aduth — APPROVED** [review]
Looks great aside from one observation noted above 👍

File: nux/store/test/reducer.js [hunk 3]
```diff
@@ -20,15 +20,22 @@ describe( 'reducer', () => {
...
 
 		it( 'should flip when tips are disabled', () => {
-			const state = areTipsDisabled( false, {
+			const state = areTipsEnabled( true, {
 				type: 'DISABLE_TIPS',
 			} );
+			expect( state ).toBe( false );
+		} );
+
+		it( 'sholud flip when tips are enabled', () => {
```
[the remark this answers is not in the document]
Comment (noisysocks) [reply nux/store/test/reducer.js]: Whoops! Fixed in b66338cfe.

**Comment by noisysocks** [comment]
> Observation: When I first loaded up the branch, the menu item showed as checked, yet I did not see any tip prompts.

I think that this is a consequence of us not marking tips as disabled once the user has dismissed every tip that we show as part of the guide. We may want to explore making it so that _all tips are disabled_ is represented in Redux in the same way as _every tip is dismissed_. Not sure how this affects third party tips, though. I've made a note to look into this 😄
